Using ObjectId property in iOS application breaks compilation

Adding a property of type `org.mongodb.kbson.ObjectId` breaks compilation of a KMM app for iOS. 

Modifying the iOS enabled `final` branch of https://github.com/Kotlin/kmm-integration-sample/tree/final by adding a `ObjectId` property like below in https://github.com/Kotlin/kmm-integration-sample/blob/final/shared/src/commonMain/kotlin/com/jetbrains/simplelogin/shared/Greeting.kt#L3
 
```
class Greeting {
    val id: ObjectId = ObjectId()
    fun greeting(): String {
        return "Hello, ${Platform().platform}!"
    }
}
```

Will make the iOS build fail with 
```
SwiftDriver simpleLoginIOS normal arm64 com.apple.xcode.tools.swift.compiler (in target 'simpleLoginIOS' from project 'SimpleLoginIOS')
    cd /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp
    builtin-SwiftDriver -- /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc -module-name simpleLoginIOS -Onone -enforce-exclusivity\=checked @/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOS.SwiftFileList -DDEBUG -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -target arm64-apple-ios15.2-simulator -enable-bare-slash-regex -g -module-cache-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -Xfrontend -serialize-debugging-options -enable-testing -index-store-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/SimpleLoginIOS-blwkamjakkhdtcgoztkhnihdbdst/Index.noindex/DataStore -swift-version 5 -I /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -emit-localized-strings -emit-localized-strings-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64 -c -j10 -enable-batch-mode -incremental -output-file-map /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOS-OutputFileMap.json -use-frontend-parseable-output -save-temps -no-color-diagnostics -serialize-diagnostics -emit-dependencies -emit-module -emit-module-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOS.swiftmodule -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-generated-files.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-own-target-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-all-target-headers.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-project-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator/include -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources-normal/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources -Xcc -DDEBUG\=1 -emit-objc-header -emit-objc-header-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOS-Swift.h -working-directory /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp -experimental-emit-module-separately

SwiftCompile normal arm64 Compiling\ simpleLoginIOSApp.swift /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/simpleLoginIOSApp.swift (in target 'simpleLoginIOS' from project 'SimpleLoginIOS')
    cd /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp
    builtin-swiftTaskExecution -- /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift -primary-file /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/simpleLoginIOSApp.swift -emit-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.d -emit-reference-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.swiftdeps -serialize-diagnostics-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.dia -emit-localized-strings -emit-localized-strings-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64 -target arm64-apple-ios15.2-simulator -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -I /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -no-color-diagnostics -enable-testing -g -module-cache-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -Onone -D DEBUG -serialize-debugging-options -enable-bare-slash-regex -empty-abi-descriptor -Xcc -working-directory -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-generated-files.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-own-target-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-all-target-headers.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-project-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator/include -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources-normal/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources -Xcc -DDEBUG\=1 -module-name simpleLoginIOS -frontend-parseable-output -disable-clang-spi -target-sdk-version 16.1 -o /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.o -index-unit-output-path /SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.o -index-store-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/SimpleLoginIOS-blwkamjakkhdtcgoztkhnihdbdst/Index.noindex/DataStore -index-system-modules

CompileSwift normal arm64 /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/simpleLoginIOSApp.swift (in target 'simpleLoginIOS' from project 'SimpleLoginIOS')
    cd /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -c /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift -primary-file /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/simpleLoginIOSApp.swift -emit-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.d -emit-reference-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.swiftdeps -serialize-diagnostics-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.dia -emit-localized-strings -emit-localized-strings-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64 -target arm64-apple-ios15.2-simulator -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -I /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -no-color-diagnostics -enable-testing -g -module-cache-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -Onone -D DEBUG -serialize-debugging-options -enable-bare-slash-regex -empty-abi-descriptor -Xcc -working-directory -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-generated-files.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-own-target-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-all-target-headers.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-project-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator/include -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources-normal/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources -Xcc -DDEBUG\=1 -module-name simpleLoginIOS -frontend-parseable-output -disable-clang-spi -target-sdk-version 16.1 -o /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.o -index-unit-output-path /SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/simpleLoginIOSApp.o -index-store-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/SimpleLoginIOS-blwkamjakkhdtcgoztkhnihdbdst/Index.noindex/DataStore -index-system-modules
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected member name or ';' after declaration specifiers
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:173:19: note: expanded from macro 'FALSE'
    #define FALSE       0
                        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:1: error: property requires fields to be named
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected ';' at end of declaration list
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
                                             ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected identifier or '('
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:173:19: note: expanded from macro 'FALSE'
    #define FALSE       0
                        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:46: error: expected member name or ';' after declaration specifiers
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:169:18: note: expanded from macro 'TRUE'
    #define TRUE        1
                        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:1: error: property requires fields to be named
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:46: error: expected ';' at end of declaration list
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
                                             ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:46: error: expected identifier or '('
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:169:18: note: expanded from macro 'TRUE'
    #define TRUE        1
                        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift:2:8: error: could not build Objective-C module 'shared'
import shared
       ^

SwiftCompile normal arm64 Compiling\ ContentView.swift /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift (in target 'simpleLoginIOS' from project 'SimpleLoginIOS')
    cd /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp
    builtin-swiftTaskExecution -- /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/simpleLoginIOSApp.swift -emit-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.d -emit-reference-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.swiftdeps -serialize-diagnostics-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.dia -emit-localized-strings -emit-localized-strings-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64 -target arm64-apple-ios15.2-simulator -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -I /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -no-color-diagnostics -enable-testing -g -module-cache-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -Onone -D DEBUG -serialize-debugging-options -enable-bare-slash-regex -empty-abi-descriptor -Xcc -working-directory -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-generated-files.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-own-target-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-all-target-headers.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-project-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator/include -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources-normal/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources -Xcc -DDEBUG\=1 -module-name simpleLoginIOS -frontend-parseable-output -disable-clang-spi -target-sdk-version 16.1 -o /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.o -index-unit-output-path /SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.o -index-store-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/SimpleLoginIOS-blwkamjakkhdtcgoztkhnihdbdst/Index.noindex/DataStore -index-system-modules

CompileSwift normal arm64 /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift (in target 'simpleLoginIOS' from project 'SimpleLoginIOS')
    cd /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -c -primary-file /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/simpleLoginIOSApp.swift -emit-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.d -emit-reference-dependencies-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.swiftdeps -serialize-diagnostics-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.dia -emit-localized-strings -emit-localized-strings-path /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64 -target arm64-apple-ios15.2-simulator -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -I /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -F /Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1 -no-color-diagnostics -enable-testing -g -module-cache-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -Onone -D DEBUG -serialize-debugging-options -enable-bare-slash-regex -empty-abi-descriptor -Xcc -working-directory -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/iosApp -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-generated-files.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-own-target-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-all-target-headers.hmap -Xcc -iquote -Xcc /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/simpleLoginIOS-project-headers.hmap -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/Debug-iphonesimulator/include -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources-normal/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources/arm64 -Xcc -I/Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/DerivedSources -Xcc -DDEBUG\=1 -module-name simpleLoginIOS -frontend-parseable-output -disable-clang-spi -target-sdk-version 16.1 -o /Users/claus.rorbech/contrib/kmm-integration-sample/build/ios/SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.o -index-unit-output-path /SimpleLoginIOS.build/Debug-iphonesimulator/simpleLoginIOS.build/Objects-normal/arm64/ContentView.o -index-store-path /Users/claus.rorbech/Library/Developer/Xcode/DerivedData/SimpleLoginIOS-blwkamjakkhdtcgoztkhnihdbdst/Index.noindex/DataStore -index-system-modules
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected member name or ';' after declaration specifiers
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:173:19: note: expanded from macro 'FALSE'
    #define FALSE       0
                        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:1: error: property requires fields to be named
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected ';' at end of declaration list
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
                                             ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected identifier or '('
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:173:19: note: expanded from macro 'FALSE'
    #define FALSE       0
                        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:46: error: expected member name or ';' after declaration specifiers
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:169:18: note: expanded from macro 'TRUE'
    #define TRUE        1
                        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:1: error: property requires fields to be named
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:46: error: expected ';' at end of declaration list
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
                                             ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/shared.h"
        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:726:46: error: expected identifier or '('
@property (readonly) SharedKbsonBsonBoolean *TRUE __attribute__((swift_name("TRUE")));
                                             ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:169:18: note: expanded from macro 'TRUE'
    #define TRUE        1
                        ^
/Users/claus.rorbech/contrib/kmm-integration-sample/iosApp/simpleLoginIOS/ContentView.swift:2:8: error: could not build Objective-C module 'shared'
import shared
       ^

** BUILD FAILED **
```

The compiler errors are already in the snippit. 
```
/Users/claus.rorbech/contrib/kmm-integration-sample/shared/build/xcode-frameworks/Debug/iphonesimulator16.1/shared.framework/Headers/shared.h:725:46: error: expected member name or ';' after declaration specifiers
@property (readonly) SharedKbsonBsonBoolean *FALSE __attribute__((swift_name("FALSE")));
```

Seems to be the constants `TRUE` and `FALSE` from https://github.com/mongodb/kbson/blob/main/src/commonMain/kotlin/org/mongodb/kbson/BsonBoolean.kt#L59..L62
that is exposed as named constants that overlaps with reserved words in Swift. @cmelchior mentioned that there might exist some way over controlling the name that it is expose as in ObjC/Swift, but haven't been able to find it. 

Added a branch of the modified sample in https://github.com/rorbech/kmm-integration-sample/tree/cr/final-with-bson-objectid for easier reproducibility. Just clone and open in Android Studio with KMM plugin and try to run on iOS.

--- a/src/commonMain/kotlin/org/mongodb/kbson/BsonBoolean.kt
+++ b/src/commonMain/kotlin/org/mongodb/kbson/BsonBoolean.kt
@@ -56,20 +56,20 @@ public class BsonBoolean(public val value: Boolean) : BsonValue(), Comparable<Bs
 
     public companion object {
         /** The true value. */
-        @JvmStatic public val TRUE: BsonBoolean = BsonBoolean(true)
+        @JvmStatic public val TRUE_VALUE: BsonBoolean = BsonBoolean(true)
 
         /** The false value. */
-        @JvmStatic public val FALSE: BsonBoolean = BsonBoolean(false)
+        @JvmStatic public val FALSE_VALUE: BsonBoolean = BsonBoolean(false)
 
         /**
          * Returns a `BsonBoolean` instance representing the specified `boolean` value.
          *
          * @param value a boolean value.
-         * @return @link if `value` is true, [BsonBoolean.FALSE] if `value` is false
+         * @return @link if `value` is true, [BsonBoolean.FALSE_VALUE] if `value` is false
          */
         @JvmStatic
         public fun valueOf(value: Boolean): BsonBoolean {
-            return if (value) TRUE else FALSE
+            return if (value) TRUE_VALUE else FALSE_VALUE
         }
     }
 }

--- a/src/commonTest/kotlin/org/mongodb/kbson/BsonArrayTest.kt
+++ b/src/commonTest/kotlin/org/mongodb/kbson/BsonArrayTest.kt
@@ -26,7 +26,7 @@ import kotlin.test.assertTrue
 
 class BsonArrayTest {
 
-    private val listOfBsonValues = listOf(BsonBoolean.TRUE, BsonBoolean.FALSE)
+    private val listOfBsonValues = listOf(BsonBoolean.TRUE_VALUE, BsonBoolean.FALSE_VALUE)
     private val bsonValue = BsonArray(listOfBsonValues)
 
     @Test
@@ -65,57 +65,57 @@ class BsonArrayTest {
         assertEquals(BsonArray(), BsonArray())
         assertEquals(bsonValue, bsonValue)
         assertNotEquals(BsonArray(), bsonValue)
-        assertNotEquals(bsonValue, BsonArray(listOf(BsonBoolean.TRUE, BsonBoolean.TRUE)))
+        assertNotEquals(bsonValue, BsonArray(listOf(BsonBoolean.TRUE_VALUE, BsonBoolean.TRUE_VALUE)))
     }
 
     @Test
     fun shouldSupportListMethods() {
-        assertTrue(bsonValue.contains(BsonBoolean.TRUE))
+        assertTrue(bsonValue.contains(BsonBoolean.TRUE_VALUE))
         assertFalse(bsonValue.contains(BsonUndefined.UNDEFINED))
 
-        assertTrue(bsonValue.containsAll(listOf(BsonBoolean.TRUE)))
-        assertFalse(bsonValue.containsAll(listOf(BsonBoolean.TRUE, BsonUndefined.UNDEFINED)))
+        assertTrue(bsonValue.containsAll(listOf(BsonBoolean.TRUE_VALUE)))
+        assertFalse(bsonValue.containsAll(listOf(BsonBoolean.TRUE_VALUE, BsonUndefined.UNDEFINED)))
 
         val iterator = bsonValue.iterator()
         assertTrue(iterator.hasNext())
-        assertEquals(BsonBoolean.TRUE, iterator.next())
+        assertEquals(BsonBoolean.TRUE_VALUE, iterator.next())
         assertTrue(iterator.hasNext())
-        assertEquals(BsonBoolean.FALSE, iterator.next())
+        assertEquals(BsonBoolean.FALSE_VALUE, iterator.next())
         assertFalse(iterator.hasNext())
 
         val listIterator = bsonValue.listIterator()
         assertFalse(listIterator.hasPrevious())
         assertTrue(listIterator.hasNext())
-        assertEquals(BsonBoolean.TRUE, listIterator.next())
-        assertEquals(BsonBoolean.TRUE, listIterator.previous())
-        assertEquals(BsonBoolean.TRUE, listIterator.next())
-        assertEquals(BsonBoolean.FALSE, listIterator.next())
+        assertEquals(BsonBoolean.TRUE_VALUE, listIterator.next())
+        assertEquals(BsonBoolean.TRUE_VALUE, listIterator.previous())
+        assertEquals(BsonBoolean.TRUE_VALUE, listIterator.next())
+        assertEquals(BsonBoolean.FALSE_VALUE, listIterator.next())
         assertTrue(listIterator.hasPrevious())
         assertFalse(listIterator.hasNext())
 
         val subListIterator = bsonValue.listIterator(1)
         assertTrue(subListIterator.hasPrevious())
         assertTrue(subListIterator.hasNext())
-        assertEquals(BsonBoolean.FALSE, subListIterator.next())
+        assertEquals(BsonBoolean.FALSE_VALUE, subListIterator.next())
         assertTrue(subListIterator.hasPrevious())
         assertFalse(subListIterator.hasNext())
 
-        assertEquals(mutableListOf<BsonValue>(BsonBoolean.FALSE), bsonValue.subList(1, 2))
+        assertEquals(mutableListOf<BsonValue>(BsonBoolean.FALSE_VALUE), bsonValue.subList(1, 2))
         assertEquals(
             2,
             org.mongodb.kbson
-                .BsonArray(listOf(BsonBoolean.TRUE, BsonBoolean.TRUE, BsonBoolean.TRUE))
-                .lastIndexOf(BsonBoolean.TRUE))
-        assertEquals(1, bsonValue.indexOf(BsonBoolean.FALSE))
+                .BsonArray(listOf(BsonBoolean.TRUE_VALUE, BsonBoolean.TRUE_VALUE, BsonBoolean.TRUE_VALUE))
+                .lastIndexOf(BsonBoolean.TRUE_VALUE))
+        assertEquals(1, bsonValue.indexOf(BsonBoolean.FALSE_VALUE))
     }
 
     @Test
     fun shouldSupportMutableListMethods() {
         val mutableList = BsonArray()
 
-        assertTrue(mutableList.add(BsonBoolean.TRUE))
+        assertTrue(mutableList.add(BsonBoolean.TRUE_VALUE))
         assertFalse(mutableList.isEmpty())
-        assertTrue(mutableList.remove(BsonBoolean.TRUE))
+        assertTrue(mutableList.remove(BsonBoolean.TRUE_VALUE))
         assertTrue(mutableList.isEmpty())
 
         assertTrue(mutableList.addAll(listOfBsonValues))
@@ -123,7 +123,7 @@ class BsonArrayTest {
         assertTrue(mutableList.removeAll(listOfBsonValues))
         assertTrue(mutableList.isEmpty())
 
-        mutableList.addAll(listOf(BsonNull, BsonBoolean.TRUE, BsonBoolean.FALSE, BsonUndefined))
+        mutableList.addAll(listOf(BsonNull, BsonBoolean.TRUE_VALUE, BsonBoolean.FALSE_VALUE, BsonUndefined))
         assertEquals(BsonNull, mutableList.removeAt(0))
         assertEquals(3, mutableList.size)
 
@@ -135,13 +135,13 @@ class BsonArrayTest {
         assertEquals(BsonNull, mutableList.set(0, BsonMinKey))
         assertEquals(BsonMinKey, mutableList.first())
 
-        val expected = listOf(BsonMinKey, BsonBoolean.TRUE, BsonBoolean.FALSE, BsonUndefined, BsonMaxKey)
+        val expected = listOf(BsonMinKey, BsonBoolean.TRUE_VALUE, BsonBoolean.FALSE_VALUE, BsonUndefined, BsonMaxKey)
         assertContentEquals(expected, mutableList.values)
 
         assertTrue(mutableList.retainAll(listOf(BsonMinKey, BsonMaxKey, BsonUndefined)))
         assertContentEquals(listOf(BsonMinKey, BsonUndefined, BsonMaxKey), mutableList.values)
 
-        assertTrue(mutableList.addAll(1, listOf(BsonBoolean.TRUE, BsonBoolean.FALSE)))
+        assertTrue(mutableList.addAll(1, listOf(BsonBoolean.TRUE_VALUE, BsonBoolean.FALSE_VALUE)))
         assertEquals(expected, mutableList.values)
 
         mutableList.clear()

--- a/src/commonTest/kotlin/org/mongodb/kbson/BsonBooleanTest.kt
+++ b/src/commonTest/kotlin/org/mongodb/kbson/BsonBooleanTest.kt
@@ -25,8 +25,8 @@ class BsonBooleanTest {
 
     @Test
     fun shouldHaveTheExpectedBsonType() {
-        assertTrue { BsonBoolean.TRUE.isBoolean() }
-        assertEquals(BsonType.BOOLEAN, BsonBoolean.TRUE.bsonType)
+        assertTrue { BsonBoolean.TRUE_VALUE.isBoolean() }
+        assertEquals(BsonType.BOOLEAN, BsonBoolean.TRUE_VALUE.bsonType)
     }
 
     @Test
@@ -37,17 +37,17 @@ class BsonBooleanTest {
 
     @Test
     fun shouldOverrideEquals() {
-        assertEquals(BsonBoolean.TRUE, BsonBoolean(true))
-        assertEquals(BsonBoolean.FALSE, BsonBoolean(false))
-        assertNotEquals(BsonBoolean.TRUE, BsonBoolean(false))
-        assertNotEquals(BsonBoolean.FALSE, BsonBoolean(true))
+        assertEquals(BsonBoolean.TRUE_VALUE, BsonBoolean(true))
+        assertEquals(BsonBoolean.FALSE_VALUE, BsonBoolean(false))
+        assertNotEquals(BsonBoolean.TRUE_VALUE, BsonBoolean(false))
+        assertNotEquals(BsonBoolean.FALSE_VALUE, BsonBoolean(true))
     }
 
     @Test
     fun shouldBeComparable() {
-        assertEquals(-1, BsonBoolean.FALSE.compareTo(BsonBoolean.TRUE))
-        assertEquals(0, BsonBoolean.TRUE.compareTo(BsonBoolean.TRUE))
-        assertEquals(0, BsonBoolean.FALSE.compareTo(BsonBoolean.FALSE))
-        assertEquals(1, BsonBoolean.TRUE.compareTo(BsonBoolean.FALSE))
+        assertEquals(-1, BsonBoolean.FALSE_VALUE.compareTo(BsonBoolean.TRUE_VALUE))
+        assertEquals(0, BsonBoolean.TRUE_VALUE.compareTo(BsonBoolean.TRUE_VALUE))
+        assertEquals(0, BsonBoolean.FALSE_VALUE.compareTo(BsonBoolean.FALSE_VALUE))
+        assertEquals(1, BsonBoolean.TRUE_VALUE.compareTo(BsonBoolean.FALSE_VALUE))
     }
 }

--- a/src/jvmTest/kotlin/org/mongodb/kbson/BsonValueApiTest.kt
+++ b/src/jvmTest/kotlin/org/mongodb/kbson/BsonValueApiTest.kt
@@ -64,7 +64,7 @@ class BsonValueApiTest {
         assertEquals(bsonConstructors, kBsonConstructors)
 
         val bsonMethods = getMethodNames(bsonClass)
-        val kBsonMethods = getMethodNames(kBsonClass, listOf("getTRUE", "getFALSE"))
+        val kBsonMethods = getMethodNames(kBsonClass, listOf("getTRUE_VALUE", "getFALSE_VALUE"))
 
         assertEquals(bsonMethods, kBsonMethods)
     }

